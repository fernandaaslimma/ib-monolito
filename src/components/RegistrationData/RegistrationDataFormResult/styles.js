import styled, { css } from "styled-components";
import { rem, media, remFontSize } from "../../../styles/tools";
import { blue70, lightBlack } from "../../../styles/settings";

export const Wrapper = styled.div`
  padding: ${rem(32)} 0 0 0;
  font-family: "Lato";
  text-align: center;
`;

export const Title = styled.h2`
  font-size: ${remFontSize(28)};
  font-family: "Lato Bold";
  letter-spacing: ${rem(0.47)};
  color: ${blue70};
  margin-top: ${rem(16)};
  font-weight: bold;
  font-stretch: normal;
  font-style: normal;
  line-height: normal;
`;

export const Message = styled.p`
  font-family: "Lato";
  font-size: ${remFontSize(18)};
  color: ${lightBlack};
  font-stretch: normal;
  font-style: normal;
  line-height: normal;
  letter-spacing: ${rem(0.48)};
  ${media.md(css`
    width: ${rem(620)};
  `)};
  margin: 16px auto 0;
`;

export const Buttons = styled.div`
  margin-top: ${rem(40)};
  display: flex;
  justify-content: center;

  > * {
    margin: ${rem(10)};
    height: auto;
    line-height: ${rem(50)};
  }

  ${media.md(css`
    margin-top: ${rem(50)};
  `)};
`;
