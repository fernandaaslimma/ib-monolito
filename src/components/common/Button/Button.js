import React from "react";
import { bool, node, string, func, number } from "prop-types";
import {
  ButtonTag,
  Content,
  WrapperChildren,
  WrapperIcon
} from "../../../styles/objects";
import LocalLoading from "../LocalLoading/";
import Icon from "../Icon";

function Button({
  children,
  onClick,
  isCallToAction,
  isWarning,
  type,
  disabled,
  width,
  className,
  small,
  height,
  dataTest,
  paddingSize,
  opacity,
  approveButton,
  actionSecondary,
  loading,
  background,
  color,
  border,
  fontSize,
  borderRadius,
  noBorderRadius,
  margin,
  padding,
  noHoverBackground,
  smallBlue,
  backgroundLoading,
  id,
  withIcon,
  isModalType,
  style
}) {
  return (
    <ButtonTag
      id={id}
      className={className}
      width={width}
      height={height}
      approveButton={approveButton}
      type={type}
      disabled={disabled}
      isCallToAction={isCallToAction}
      isWarning={isWarning}
      data-test={dataTest ? dataTest : Button.displayName}
      onClick={onClick}
      small={small}
      smallBlue={smallBlue}
      opacity={opacity}
      paddingSize={paddingSize}
      actionSecondary={actionSecondary}
      loading={loading}
      background={background}
      color={color}
      border={border}
      fontSize={fontSize}
      borderRadius={borderRadius}
      noBorderRadius={noBorderRadius}
      margin={margin}
      padding={padding}
      noHoverBackground={noHoverBackground}
      isModalType={isModalType}
      style={style}
    >
      <Content>
        {withIcon && !loading && (
          <WrapperIcon>
            <Icon
              type={withIcon.name}
              width={withIcon.width}
              height={withIcon.height}
            ></Icon>
          </WrapperIcon>
        )}
        {loading ? (
          <LocalLoading background={backgroundLoading} />
        ) : (
          <WrapperChildren withIcon={withIcon}>{children}</WrapperChildren>
        )}
      </Content>
    </ButtonTag>
  );
}

Button.displayName = "Button";

Button.defaultProps = {
  isCallToAction: null,
  isWarning: null,
  type: "button",
  disabled: false,
  width: null,
  height: null,
  onClick: null,
  className: null,
  small: false,
  paddingSize: null,
  actionSecondary: null
};

Button.propTypes = {
  children: node,
  isCallToAction: bool,
  isWarning: bool,
  disabled: bool,
  type: string,
  width: number,
  onClick: func,
  className: string,
  small: bool,
  height: number,
  paddingSize: number,
  actionSecondary: bool
};

export default Button;
