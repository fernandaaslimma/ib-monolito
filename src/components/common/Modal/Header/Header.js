import React, { Component } from "react";
import {
  Wrapper,
  HeaderTitle,
  CloseIcon,
  BackButton,
  StyledCloseIcon
} from "./styles";
import { Button, Icon } from "react-bocombbm-components";
import { translate } from "../../../../utils/i18n";

class Header extends Component {
  constructor(props) {
    super(props);

    this.state = {};
  }

  render() {
    const {
      headerOnClickBack,
      headerOnClickClose,
      onClickBack,
      onClickClose,
      title,
      removeCloseIcon = false
    } = this.props;
    const onClickBackFunction = headerOnClickBack || onClickBack || false;
    const onClickCloseFunction = headerOnClickClose || onClickClose || false;
    return (
      <Wrapper wrapperStyle={this.props.wrapperStyle}>
        {onClickBackFunction && (
          <BackButton>
            <Button
              dataTest="backButton"
              withIcon={{ name: "IconBack", position: "left" }}
              type="text"
              small
              onClick={onClickBackFunction}
            >
              {translate("BACK")}
            </Button>
          </BackButton>
        )}
        <HeaderTitle bottomSheetHeaderStyle={this.props.bottomSheetHeaderStyle}>
          {title || "\u00A0"}
        </HeaderTitle>
        {removeCloseIcon === false &&
          (onClickCloseFunction && !this.props.bottomSheetHeaderStyle ? (
            <CloseIcon
              type="Close"
              dataTest="registration-data-close"
              width="20"
              height="20"
              onClick={onClickCloseFunction}
            />
          ) : onClickCloseFunction && this.props.bottomSheetHeaderStyle ? (
            <StyledCloseIcon>
              <Icon
                data-test="closeKnowMoreButton"
                type="BorderedClose"
                onClick={onClickCloseFunction}
              />{" "}
            </StyledCloseIcon>
          ) : null)}
      </Wrapper>
    );
  }
}
export default Header;
