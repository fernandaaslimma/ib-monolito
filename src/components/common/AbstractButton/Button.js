import React from "react";
import { bool, node, string, func } from "prop-types";
import { ButtonTag } from "./styles";

function Button({
  children,
  onClick,
  isCallToAction,
  isWarning,
  isSecondary,
  isSpanButton,
  type,
  disabled,
  width,
  className,
  small,
  dataTest,
  opacity,
  height,
  noPadding,
  heightPx
}) {
  return (
    <ButtonTag
      className={className}
      width={width}
      type={type}
      disabled={disabled}
      isCallToAction={isCallToAction}
      isWarning={isWarning}
      isSecondary={isSecondary}
      isSpanButton={isSpanButton}
      data-test={dataTest ? dataTest : Button.displayName}
      onClick={onClick}
      small={small}
      opacity={opacity}
      height={height}
      noPadding={noPadding}
      heightPx={heightPx}
    >
      {children}
    </ButtonTag>
  );
}

Button.displayName = "Button";

Button.defaultProps = {
  isCallToAction: null,
  isWarning: null,
  type: "button",
  disabled: false,
  width: null,
  onClick: null,
  className: null,
  small: false
};

Button.propTypes = {
  children: node.isRequired, // eslint-disable-line
  isCallToAction: bool,
  isWarning: bool,
  isSecondary: bool,
  isSpanButton: bool,
  disabled: bool,
  type: string,
  width: string,
  onClick: func,
  className: string,
  small: bool,
  dataTest: string,
  opacity: string,
  height: string,
  noPadding: bool,
  heightPx: string
};

export default Button;
