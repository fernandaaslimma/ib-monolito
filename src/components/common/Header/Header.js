/* eslint-disable react/display-name */
import React, { Fragment, useContext, useEffect, useState } from "react";
import {
  UserContextMenu,
  Header as Head,
  NavigationMenu
} from "react-bocombbm-components";
import {
  LoaderArea,
  ChangeLanguageContainer,
  DesktopContent,
  NavigationMenuWrapper,
  HeadWrapper,
  PagenameContainer,
  PageMenu,
  PageSubMenu
} from "./styles";

import { translate } from "../../../utils/i18n";
import ChangeLanguage from "../ChangeLanguage";
import { MODAL_TYPES } from "../Modal/Modal";

import { redirect } from "../../../utils/redirect";
import { checkViewContextAndRedirect } from "../../../utils/fetchHandler";
import { checkIfHasAccess, checkUserType } from "../CanAccess/CanAccess";
import Hide from "../../common/Hide";

import {
  GET_TRANSACTIONS,
  GET_POSITION,
  GET_CONTRACT,
  GET_DOCUMENTS,
  CREATE_EFT,
  CREATE_APPROVE_EFT,
  APPROVE_EFT,
  GET_EFT,
  GET_PERSON_REGISTRATION,
  GET_TERMS,
  GET_NOTIFICATIONS,
  GET_STATEMENT,
  CORPORATION,
  INDIVIDUAL,
  INVESTMENT_PRODUCTS_LIST_URL,
  INVESTMENT_FUNDS_ROLE,
  INVESTMENT_FIXED_INCOME_ROLE,
  GET_CONSENTS,
  NEW_CONSENTS,
  SHIPMENTS_RETURN,
  SHIPMENTS_UPLOAD,
  VALIDATION_FILE,
  RECEIVABLES,
  INTERNATIONAL_TRANSFER,
  PRIVATEACCOUNT,
  TRANSACTIONALACCOUNT,
  GET_SUITABILITY_FORM,
  GET_RECEIVABLES_REPORTS
} from "../../../utils/constants";
import Menu from "../Menu";
import { rem } from "../../../styles/tools";
import history from "../../../services/history";
import LogoutContent from "./LogoutContent";
import { Context } from "../OffshoreSelect/offshoreContext";

export function userName(userInfo) {
  if (userInfo && userInfo.givenName) {
    return userInfo.givenName;
  }
  return translate("USER");
}

const Header = ({
  doLogout,
  openModal,
  closeModal,
  getAccounts,
  userInfo,
  isNavigationMenuShown,
  showNavigationMenu,
  accounts,
  openToastr,
  getInvestorProfile,
  investorProfile
}) => {
  const [currentMenuLabel, setCurrentMenuLabel] = useState("");
  const [currentSubMenuLabel, setCurrentSubMenuLabel] = useState("");
  const [config, setConfig] = useState([]);
  const [avatarConfig, setAvatarConfig] = useState([]);

  const { isGlobalMode } = useContext(Context);

  const logoutHandler = () => {
    openModal({
      type: MODAL_TYPES.PAGE_AS_MODAL,
      width: "auto",
      children: () => (
        <LogoutContent
          doLogout={() =>
            doLogout().then(() => {
              checkViewContextAndRedirect();
            })
          }
          closeModal={() => closeModal()}
        />
      )
    });
  };

  const mobileAditionalAction = {
    label: translate("ACCOUNT_LOGOUT"),
    icon: "Logout",
    action: logoutHandler
  };

  const redirectTo = path => {
    redirect(path);
  };

  const setGlobalMenu = () => {
    setConfig([
      {
        label: userInfo.givenName,
        icon: "Avatar",
        allowed: true,
        mobileOnly: true,
        routeLink: "/user_menu",
        subRouteLink: "/suitability",
        subtitle: investorProfile && translate("SUITABILITY_VIEW_PROFILE"),
        profileLabel: investorProfile,
        secondLevelLabel: translate("MY_PROFILE"),
        style: { padding: rem(16), marginTop: rem(24) },
        bankName: "107 - BOCOM BBM SA",
        accountNumber: accounts.length > 0 ? accounts[0].number : "",
        accountBranch: accounts.length > 0 ? accounts[0].branch : "",
        accountVerifyingDigit:
          accounts.length > 0 ? accounts[0].verifyingDigit : "",
        subItems: [
          {
            label: translate("REGISTRATION_DATA"),
            icon: "RegisterData",
            allowed: checkIfHasAccess(userInfo, [GET_PERSON_REGISTRATION]),
            alternativeLink: "/registrationData",
            routeLink: "/user_menu/test"
          },
          {
            label: translate("INVESTOR_PROFILE"),
            icon: "InvestorProfile",
            alternativeLink: "/suitability",
            routeLink: "/user_menu/test2",
            allowed: checkIfHasAccess(userInfo, [
              GET_NOTIFICATIONS,
              GET_SUITABILITY_FORM
            ])
          },
          {
            label: translate("TERMS"),
            icon: "MenuReports",
            allowed: checkIfHasAccess(userInfo, [GET_TERMS]),
            alternativeLink: "/terms",
            routeLink: "/user_menu/test3"
          }
        ]
      },
      {
        label: translate("CASH_ACCOUNT"),
        routeLink: "/cashaccounts/statements",
        dataTest: "Navbar_CashAccounts",
        icon: "MenuCashAccount",
        style: { marginTop: rem(16) },
        allowed: checkIfHasAccess(userInfo, [GET_STATEMENT])
      },
      {
        label: translate("INVESTMENTS"),
        routeLink: "/investments",
        dataTest: "Navbar_Investments",
        icon: "MenuInvestments",
        allowed: checkIfHasAccess(
          userInfo,
          [GET_TRANSACTIONS, GET_POSITION],
          true
        ),
        subItems: [
          {
            label: translate("OVERVIEW"),
            routeLink: "/investments/overview",
            dataTest: "SubnavTab-0",
            icon: "SubMenuOverView",
            allowed: checkIfHasAccess(userInfo, [
              GET_TRANSACTIONS,
              GET_POSITION
            ])
          },
          {
            label: translate("POSITIONS"),
            routeLink: "/investments/positions",
            dataTest: "SubnavTab-1",
            icon: "SubMenuPosition",
            allowed: checkIfHasAccess(userInfo, [GET_POSITION])
          }
        ]
      }
    ]);
  };

  const setMenuBrasil = () => {
    setConfig([
      {
        label: userInfo.givenName,
        icon: "Avatar",
        allowed: true,
        mobileOnly: true,
        routeLink: "/user_menu",
        subRouteLink: "/suitability",
        subtitle: investorProfile && translate("SUITABILITY_VIEW_PROFILE"),
        profileLabel: investorProfile,
        secondLevelLabel: translate("MY_PROFILE"),
        style: { padding: rem(16), marginTop: rem(24) },
        bankName: "107 - BOCOM BBM SA",
        accountNumber: accounts.length > 0 ? accounts[0].number : "",
        accountBranch: accounts.length > 0 ? accounts[0].branch : "",
        accountVerifyingDigit:
          accounts.length > 0 ? accounts[0].verifyingDigit : "",
        subItems: [
          {
            label: translate("REGISTRATION_DATA"),
            icon: "RegisterData",
            allowed: checkIfHasAccess(userInfo, [GET_PERSON_REGISTRATION]),
            alternativeLink: "/registrationData",
            routeLink: "/user_menu/test"
          },
          {
            label: translate("INVESTOR_PROFILE"),
            icon: "InvestorProfile",
            alternativeLink: "/suitability",
            routeLink: "/user_menu/test2",
            allowed: checkIfHasAccess(userInfo, [
              GET_NOTIFICATIONS,
              GET_SUITABILITY_FORM
            ])
          },
          {
            label: translate("TERMS"),
            icon: "MenuReports",
            allowed: checkIfHasAccess(userInfo, [GET_TERMS]),
            alternativeLink: "/terms",
            routeLink: "/user_menu/test3"
          }
        ]
      },
      {
        label: translate("CASH_ACCOUNT"),
        routeLink: "/cashaccounts",
        dataTest: "Navbar_CashAccounts",
        icon: "MenuCashAccount",
        style: { marginTop: rem(16) },
        allowed: checkIfHasAccess(
          userInfo,
          [CREATE_EFT, CREATE_APPROVE_EFT, APPROVE_EFT, GET_EFT, GET_STATEMENT],
          true
        ),
        subItems: [
          {
            label: translate("TRANSFER"),
            routeLink: "/cashaccounts/dashboard",
            icon: "SubMenuTransfers",
            dataTest: "SubnavTab-0",
            allowed:
              checkIfHasAccess(
                userInfo,
                [CREATE_EFT, CREATE_APPROVE_EFT, APPROVE_EFT, GET_EFT],
                true
              ) && checkUserType(userInfo, CORPORATION)
          },
          {
            label: translate("TRANSFER"),
            routeLink: "/cashaccounts/wiretransfer",
            icon: "SubMenuTransfers",
            dataTest: "SubnavTab-1",
            allowed:
              checkIfHasAccess(
                userInfo,
                [CREATE_EFT, CREATE_APPROVE_EFT, APPROVE_EFT, GET_EFT],
                true
              ) && checkUserType(userInfo, INDIVIDUAL)
          },
          {
            label: translate("STATEMENTS"),
            routeLink: "/cashaccounts/statements",
            dataTest: "subnavTabStatements",
            icon: "SubMenuStatements",
            allowed: checkIfHasAccess(userInfo, [GET_STATEMENT])
          }
        ]
      },
      {
        label: translate("INVESTMENTS"),
        routeLink: "/investments",
        dataTest: "Navbar_Investments",
        icon: "MenuInvestments",
        allowed: checkIfHasAccess(
          userInfo,
          [GET_TRANSACTIONS, GET_POSITION],
          true
        ),
        subItems: [
          {
            label: translate("OVERVIEW"),
            routeLink: "/investments/overview",
            dataTest: "SubnavTab-0",
            icon: "SubMenuOverView",
            allowed: checkIfHasAccess(userInfo, [
              GET_TRANSACTIONS,
              GET_POSITION
            ])
          },
          {
            label: translate("POSITIONS"),
            routeLink: "/investments/positions",
            dataTest: "SubnavTab-1",
            icon: "SubMenuPosition",
            allowed: checkIfHasAccess(userInfo, [GET_POSITION]),
            subItems: [
              {
                label: translate("FIXED_INCOME"),
                routeLink: "/investments/positions/fixed-income",
                dataTest: "ThirdnavTab-0",
                allowed: true,
                icon: "SubMenuFixedIncome"
              },
              {
                label: translate("EQUITIES"),
                routeLink: "/investments/positions/equities",
                dataTest: "ThirdnavTab-1",
                allowed: true,
                icon: "SubMenuEquities"
              },
              {
                label: translate("FUNDS"),
                routeLink: "/investments/positions/funds",
                dataTest: "ThirdnavTab-2",
                allowed: true,
                icon: "SubMenuFunds"
              }
            ]
          },
          {
            label: translate("INVESTMENT_PRODUCTS"),
            routeLink: INVESTMENT_PRODUCTS_LIST_URL,
            icon: "MenuInvestments",
            allowed: checkIfHasAccess(
              userInfo,
              [INVESTMENT_FUNDS_ROLE, INVESTMENT_FIXED_INCOME_ROLE],
              true
            ),
            dataTest: "SubnavTab-3"
          }
        ]
      },
      {
        label: translate("FOREIGN_EXCHANGE"),
        routeLink: "/exchanges",
        dataTest: "Navbar_Exchanges",
        icon: "MenuExchange",
        allowed: checkIfHasAccess(userInfo, [GET_CONTRACT]),
        subItems: [
          {
            label: translate("PENDING_CONTRACTS"),
            routeLink: "/exchanges/unsigned",
            icon: "SubMenuPendingContracts",
            dataTest: "PendingContracts",
            allowed: true
          },
          {
            label: translate("CONTRACT_HISTORY"),
            routeLink: "/exchanges/signed",
            icon: "MenuReports",
            dataTest: "ContractHistory",
            allowed: true
          },
          {
            label: translate("INTERNATIONAL_TRANSFER"),
            routeLink: "/exchanges/international-transfer",
            icon: "SubMenuInternationalTransfer",
            dataTest: "InternationalTransfer",
            allowed:
              checkIfHasAccess(userInfo, [INTERNATIONAL_TRANSFER]) &&
              checkUserType(userInfo, INDIVIDUAL)
          }
        ]
      },
      {
        label: translate("REPORTS"),
        routeLink: "/reports",
        dataTest: "Navbar_Reports",
        icon: "MenuReports",
        style: { marginTop: rem(24) },
        allowed: checkIfHasAccess(userInfo, [GET_DOCUMENTS])
      },
      {
        label: translate("REMITTANCES_CREDIT"),
        routeLink: "/shipments",
        icon: "MenuShipments",
        dataTest: "Navbar_shipments",
        allowed: checkIfHasAccess(userInfo, [RECEIVABLES]),
        subItems: [
          {
            label: translate("REMITTANCES_UPLOAD_HEADER_SUB_ITEM"),
            routeLink: "/shipments/upload",
            icon: "SubMenuRemittance",
            dataTest: "Sub_Navbar_Remittances_Upload",
            allowed: checkIfHasAccess(userInfo, [SHIPMENTS_UPLOAD])
          },
          {
            label: translate("REMITTANCES_DOWNLOAD_HEADER_SUB_ITEM"),
            routeLink: "/shipments/returns",
            icon: "SubMenuReturns",
            dataTest: "Sub_Navbar_Remittances_Returns",
            allowed: checkIfHasAccess(userInfo, [SHIPMENTS_RETURN])
          },
          {
            label: translate("REPORTS"),
            routeLink: "/shipments/report",
            icon: "SubMenuRemittance",
            dataTest: "Sub_Navbar_Remittances_Reports",
            allowed: checkIfHasAccess(userInfo, [GET_RECEIVABLES_REPORTS])
          },
          {
            label: translate("REMITTANCES_UPLOAD_VALIDATION_MENO"),
            routeLink: "/shipments/validation",
            icon: "SubMenuFileValidation",
            dataTest: "Sub_Navbar_File_Validation",
            allowed: checkIfHasAccess(userInfo, [VALIDATION_FILE])
          }
        ]
      },
      {
        label: translate("OPEN_BANKING"),
        routeLink: "/open-banking",
        dataTest: "Navbar_Open_Banking",
        icon: "MenuOpenFinance",
        allowed: checkIfHasAccess(userInfo, [GET_CONSENTS]),
        subItems: [
          {
            label: translate("OPEN_BANKING_SHARES_MENU"),
            routeLink: "/open-banking/my-shares",
            icon: "SubMenuShares",
            dataTest: "Open_Banking_My_Shares",
            allowed: checkIfHasAccess(userInfo, [GET_CONSENTS])
          },
          {
            label: translate("OPEN_BANKING_NEW_CONSENT_MENU"),
            routeLink: "/open-banking/new-consent",
            icon: "SubMenuNewConsent",
            dataTest: "Open_Banking_New_Consent",
            allowed: checkIfHasAccess(userInfo, [NEW_CONSENTS])
          }
        ]
      }
    ]);
  };

  const setConstants = () => {
    if (isGlobalMode) {
      setGlobalMenu();
    } else {
      setMenuBrasil();
    }
  };

  useEffect(() => {
    if (accounts) {
      setConstants();
    }
  }, [accounts, investorProfile, isGlobalMode]);

  const getInitialInfo = async () => {
    if (checkIfHasAccess(userInfo, [GET_NOTIFICATIONS, GET_SUITABILITY_FORM])) {
      await getInvestorProfile();
    }
    getAccounts(null, false, [PRIVATEACCOUNT, TRANSACTIONALACCOUNT]);
  };

  useEffect(() => {
    if (userInfo.roles) {
      getInitialInfo();
    }

    if (userInfo?.tenants?.length <= 0 && userInfo?.roles === null) {
      setAvatarConfig([{ mobileAditionalAction }]);
    }
  }, [userInfo]);

  useEffect(() => {
    if (config.length > 0) {
      getRouteName(history.location.pathname);

      setAvatarConfig([
        {
          label: translate("REGISTRATION_DATA"),
          icon: "Person",
          routeLink: "/registrationData",
          allowed: checkIfHasAccess(userInfo, [GET_PERSON_REGISTRATION]),
          dataTest: "userContextMyData"
        },
        {
          label: translate("INVESTOR_PROFILE"),
          icon: "Suitability",
          routeLink: "/suitability",
          allowed: checkIfHasAccess(userInfo, [GET_NOTIFICATIONS]),
          dataTest: "userContextSuitability"
        },
        {
          label: translate("TERMS"),
          icon: "Terms",
          routeLink: "/terms",
          allowed: checkIfHasAccess(userInfo, [GET_TERMS]),
          dataTest: "userContextTerms"
        },
        {
          label: translate("ACCOUNT_LOGOUT"),
          icon: "Logout",
          alternFunc: logoutHandler,
          dataTest: "userLogout"
        }
      ]);
    }
  }, [config]);

  useEffect(() => {
    if (avatarConfig.length > 0) {
      if (avatarConfig.filter(item => item.allowed === true).length === 0) {
        setAvatarConfig([
          {
            label: translate("ACCOUNT_LOGOUT"),
            icon: "Logout",
            alternFunc: logoutHandler,
            dataTest: "userLogout"
          }
        ]);
      }

      if (
        config?.[0]?.subItems?.filter(item => item.allowed === true).length ===
        0
      ) {
        const newConfig = config;
        delete newConfig[0].subItems;
        setConfig(newConfig);
      }
    }
  }, [avatarConfig]);

  useEffect(() => {
    getRouteName(history.location.pathname);
  }, [history.location.pathname]);

  const getRouteName = route => {
    const menu = config.find(el => el.routeLink === route);

    if (menu) {
      setCurrentMenuLabel(menu.secondLevelLabel || menu.label);
      setCurrentSubMenuLabel("");
    } else {
      config.forEach(item => {
        if (item.subItems) {
          const subMenu = item.subItems.find(
            subItem =>
              subItem.routeLink === route || subItem.alternativeLink === route
          );
          if (subMenu) {
            setCurrentMenuLabel(item.secondLevelLabel || item.label);
            setCurrentSubMenuLabel(subMenu.label);
          }
        }
      });
    }
  };

  return (
    <Fragment>
      <Hide below="md">
        <LoaderArea />
      </Hide>
      <Hide below="md" dataTest="hideComponent">
        <HeadWrapper>
          <Head data-test="Header">
            <ChangeLanguageContainer>
              <ChangeLanguage />
            </ChangeLanguageContainer>
            <Hide below="md">
              <DesktopContent>
                <UserContextMenu
                  userInfo={{ name: userName(userInfo) }}
                  avatarConfig={avatarConfig}
                  goToRoute={redirectTo}
                />
              </DesktopContent>
            </Hide>
          </Head>
        </HeadWrapper>
      </Hide>
      <Fragment>
        <Hide above="md">
          {!isNavigationMenuShown && currentMenuLabel && (
            <PagenameContainer>
              <PageMenu currentSubMenuLabel={currentSubMenuLabel}>
                {currentMenuLabel}
              </PageMenu>
              <PageSubMenu>{currentSubMenuLabel}</PageSubMenu>
            </PagenameContainer>
          )}
        </Hide>
        <Hide below="md">
          <NavigationMenuWrapper>
            <NavigationMenu
              config={config}
              goToRoute={redirectTo}
              dataTest="navigationMenu"
              hasHeader
              currentRoute={history.location.pathname}
              mobileAditionalAction={mobileAditionalAction}
            />
          </NavigationMenuWrapper>
        </Hide>
        <Hide above="md">
          <Menu
            showNavigationMenu={showNavigationMenu}
            isNavigationMenuShown={isNavigationMenuShown}
            config={config}
            goToRoute={redirectTo}
            dataTest="navigationMenu"
            currentRoute={history.location.pathname}
            mobileAditionalAction={mobileAditionalAction}
            openToastr={openToastr}
          />
        </Hide>
      </Fragment>
    </Fragment>
  );
};

export default Header;
