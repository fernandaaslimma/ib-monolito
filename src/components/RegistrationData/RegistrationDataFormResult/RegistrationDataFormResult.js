import React, { Fragment } from "react";
import { func } from "prop-types";
import { Title, Message, Wrapper } from "./styles";

import { translate } from "../../../utils/i18n";
import { REGISTRATION_DATA_NOTIFICATION_TYPE } from "../../../utils/constants";
import { redirect } from "../../../utils/redirect";

import Header from "../../common/Modal/Header";
import { Icon, Button } from "react-bocombbm-components";

export function finish(closeModal, setNotificationStatus) {
  setNotificationStatus(REGISTRATION_DATA_NOTIFICATION_TYPE);
  closeModal();
  redirect("/home");
}

function RegistrationDataFormResult({ closeModal, setNotificationStatus }) {
  return (
    <Fragment>
      <Header onClickClose={() => finish(closeModal, setNotificationStatus)} />
      <Wrapper>
        <Icon
          data-test="ImgDocumentWaiting"
          type="DocumentChecked"
          width="192"
          height="193"
        />
        <Title>{translate("ATUCAD_DATA_SUCCESSFULLY_SENT")}</Title>
        <Message>{translate("ATUCAD_DATA_SUCCESSFULLY_SENT_MESSAGE")}</Message>
        <Button
          dataTest="registration-data-form-close-modal"
          type="conclusive"
          spacing={{ left: "none", top: "l", right: "none", bottom: "m" }}
          onClick={() => finish(closeModal, setNotificationStatus)}
        >
          {translate("ATUCAD_BACK_TO_NAVIGATION")}
        </Button>
      </Wrapper>
    </Fragment>
  );
}

RegistrationDataFormResult.propTypes = {
  closeModal: func.isRequired,
  setNotificationStatus: func.isRequired
};

export default RegistrationDataFormResult;
