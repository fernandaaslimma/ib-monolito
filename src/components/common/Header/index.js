import { connect } from "redux-zero/react";

import Header from "./Header";
import logoutActions from "../../../actions/logout";
import loginActions from "../../../actions/login";
import modalActions from "../../../actions/modal";
import navigationMenu from "../../../actions/navigationMenu";
import positionActions from "../../../actions/position";
import toastrActions from "../../../actions/toastr";
import suitabilityActions from "../../../actions/suitability";
import { combineActions } from "redux-zero/utils";

export default connect(
  ({
    userInfo,
    isNavigationMenuShown,
    showNavigationMenu,
    accounts,
    getAccounts,
    getSuitabilityInfo,
    openToastr,
    openModal,
    closeModal,
    getInvestorProfile,
    investorProfile
  }) => ({
    userInfo,
    isNavigationMenuShown,
    showNavigationMenu,
    accounts,
    getAccounts,
    getSuitabilityInfo,
    openToastr,
    openModal,
    closeModal,
    getInvestorProfile,
    investorProfile
  }),
  combineActions(
    logoutActions,
    modalActions,
    navigationMenu,
    positionActions,
    toastrActions,
    suitabilityActions,
    loginActions
  )
)(Header);
