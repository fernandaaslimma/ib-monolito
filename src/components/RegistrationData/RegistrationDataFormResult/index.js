import { connect } from "redux-zero/react";
import { combineActions } from "redux-zero/utils";

import modalActions from "../../../actions/modal";
import RegistrationDataFormResult from "./RegistrationDataFormResult";
import notificationActions from "../../../actions/notification";

export default connect(
  null,
  combineActions(modalActions, notificationActions)
)(RegistrationDataFormResult);
