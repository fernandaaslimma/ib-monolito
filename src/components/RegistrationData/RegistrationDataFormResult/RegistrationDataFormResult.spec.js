import React from "react";
import { shallow } from "enzyme";
import * as redirect from "../../../utils/redirect";

import RegistrationDataFormResult from "./RegistrationDataFormResult";
import Button from "react-bocombbm-components/dist/Button";
import Header from "../../common/Modal/Header";

jest.mock("../../../utils/i18n", () => ({
  translate: id => id,
  getLanguage: () => "en-US",
  isZhCN: () => false
}));

const props = {
  closeModal: jest.fn(),
  setNotificationStatus: jest.fn()
};

describe("Registration Data Form Result component", () => {
  it("should match snapshot", () => {
    expect(
      shallow(<RegistrationDataFormResult {...props} />)
    ).toMatchSnapshot();
  });

  it("should close modal and redirect to /home when clicking on close button", () => {
    const redirectSpy = jest.spyOn(redirect, "redirect");
    const wrapper = shallow(<RegistrationDataFormResult {...props} />);
    const close = wrapper.find(Button);

    close.simulate("click");

    expect(props.closeModal).toHaveBeenCalled();
    expect(redirect.redirect.calledOnce);
    expect(redirectSpy).toHaveBeenCalledWith("/home");
  });

  it("should close modal and redirect to /home when clicking on close button", () => {
    const redirectSpy = jest.spyOn(redirect, "redirect");
    const wrapper = shallow(<RegistrationDataFormResult {...props} />);
    const close = wrapper.find(Header);

    close.prop("onClickClose")();

    expect(props.closeModal).toHaveBeenCalled();
    expect(redirect.redirect.calledOnce);
    expect(redirectSpy).toHaveBeenCalledWith("/home");
  });
});
