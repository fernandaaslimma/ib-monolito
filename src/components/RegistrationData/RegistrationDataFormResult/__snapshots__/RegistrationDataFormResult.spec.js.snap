// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration Data Form Result component should match snapshot 1`] = `
<Fragment>
  <ConnectWrapper
    onClickClose={[Function]}
  />
  <styled.div>
    <Icon
      color={null}
      data-test="ImgDocumentWaiting"
      height="193"
      spacing={
        Object {
          "bottom": "none",
          "left": "none",
          "right": "none",
          "top": "none",
        }
      }
      type="DocumentChecked"
      width="192"
    />
    <styled.h2>
      ATUCAD_DATA_SUCCESSFULLY_SENT
    </styled.h2>
    <styled.p>
      ATUCAD_DATA_SUCCESSFULLY_SENT_MESSAGE
    </styled.p>
    <Button
      dataTest="registration-data-form-close-modal"
      disabled={false}
      onClick={[Function]}
      spacing={
        Object {
          "bottom": "m",
          "left": "none",
          "right": "none",
          "top": "l",
        }
      }
      type="conclusive"
    >
      ATUCAD_BACK_TO_NAVIGATION
    </Button>
  </styled.div>
</Fragment>
`;
